src/EditorFeatures/Core.Wpf/Preview/PreviewFactoryService.cs

```diff
@@ -585,22 +600,22 @@ public Task<DifferenceViewerPreview> CreateChangedAnalyzerConfigDocumentPreviewV
                 cancellationToken);
         }
 
-        private Task<DifferenceViewerPreview> CreateChangedDocumentViewAsync(ITextBuffer oldBuffer, ITextBuffer newBuffer, string description,
+        private async Task<DifferenceViewerPreview> CreateChangedDocumentViewAsync(ITextBuffer oldBuffer, ITextBuffer newBuffer, string description,
             List<LineSpan> originalSpans, List<LineSpan> changedSpans, PreviewWorkspace leftWorkspace, PreviewWorkspace rightWorkspace,
             double zoomLevel, CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
why are we not doing this anymore?  seems totaly reasonably if we were canceled.

Reply:
It's now part of `SwitchToMainThreadAsync` a few lines below (behavior was changed for 16.5 to make it always throw if cancellation was requested).

Reply:
ah, nice.